onchain/wallet/wallet.go

```diff
@@ -308,33 +345,23 @@ func (wallet *Wallet) GetSubaccounts(refresh bool) ([]*Subaccount, error) {
 		return nil, err
 	}
 
-	if refresh {
-		for _, subaccount := range result.Subaccounts {
-			// wait for subaccount to be synced
-			logger.Debugf("%+v", subaccount)
-			if subaccount.Used {
-				var result struct {
-					Transactions []any `json:"transactions"`
-				}
-				timeout := time.After(20 * time.Second)
-				ticker := time.NewTicker(500 * time.Millisecond)
-				defer ticker.Stop()
-				for len(result.Transactions) == 0 {
-					details, free = toJson(map[string]any{"subaccount": subaccount.Pointer, "first": 0, "count": 1})
-					err := withAuthHandler(C.GA_get_transactions(wallet.session, details, &handler), handler, &result)
-					free()
-					if err != nil {
-						return nil, fmt.Errorf("could not get transactions for subaccount %d: %w", subaccount.Pointer, err)
-					}
-					select {
-					case <-ticker.C:
-					case <-timeout:
-						return nil, fmt.Errorf("timed out waiting for subaccount %d to sync", subaccount.Pointer)
-					}
-				}
-			}
+	var newAccounts []uint64
+	for _, account := range result.Subaccounts {
+		if !slices.Contains(wallet.syncedAccounts, account.Pointer) {
+			newAccounts = append(newAccounts, account.Pointer)
+			wallet.syncedAccounts = append(wallet.syncedAccounts, account.Pointer)
 		}
 	}
+	if len(newAccounts) > 0 {
+		done := make(chan bool)
```

Comment:
This channel is never closed

Reply:
is this a problem? gc picks it up even if it doesnt get closed